### PR TITLE
add * wildcard to synchronize all tables in database

### PR DESCRIPTION
**Pull request by perfectacle**
I'm golang newbie
so my code isn't good

### CODE REVIEW

File: river/river.go [hunk 1]
```diff
@@ -140,6 +140,13 @@ func (r *River) parseSource() (map[string][]string, error) {
 
 	// first, check sources
 	for _, s := range r.c.Sources {
+		if len(s.Tables) > 1 {
+			for _, table := range s.Tables {
+				if table == "*" {
+					return nil, errors.Errorf("wildcard * not allowed for multiple tables")
```
Comment (siddontang) [inline river/river.go]: is not allowed

**Comment by siddontang** [comment]
Thanks @perfectacle 

Can you add some tests and add the doc for this feature?

File: river/river.go (hunk 1 above)
Comment (perfectacle) [reply river/river.go]: wildcard * not allowed for multiple tables, right...?

File: README.md [hunk 2]
```diff
@@ -46,6 +46,16 @@ tables = ["t3", t4]
 
 `schema` is the database name, and `tables` includes the table need to be synced.
 
+If you want to sync **all table in database**, you can use **asterisk(\*)**.  
+```
+[[source]]
+schema = "test"
+tables = ["*"]
+
+# When using an asterisk, it is not possible to sync multiple tables
```
Comment (siddontang) [inline README.md]: not allowed to

File: README.md [hunk 3]
```diff
@@ -46,6 +46,16 @@ tables = ["t3", t4]
...
+# tables = ["*", "table"] (X)
```
Comment (siddontang) [inline README.md]: remove `(x)`

**Comment by perfectacle** [comment]
hey I'm golang newbie, and newbie developer...
so make test code is hard to me...
I don't know how can I make integration test code... sorry...

**Comment by siddontang** [comment]
PTAL @EagleChen

**Comment by perfectacle** [comment]
@siddontang I have a some questions about test code...
I'm tried to make test code,
but I can't test... because I typed below command,  
```bash
go test ./river/river_test.go
```
but error message is comming  
```bash
# command-line-arguments
./river_test.go:25:5: undefined: River
FAIL    command-line-arguments [build failed]
```

How can I fix it...?

and I want to test if it is the same string...
so I use Assert in `./vendor/github.com/pingcap/check/helpers.go`  
but second parmeter(checker) provides IsTrue, IsFalse, IsNil...
how can I test about string...?

plz help me...!

**Comment by siddontang** [comment]
you can run `make test` directly.

**Comment by perfectacle** [comment]
@siddontang hey, finally I made test code, plz check it!

File: river/river_test.go [hunk 4]
```diff
@@ -350,4 +361,7 @@ func (s *riverTestSuite) TestRiver(c *C) {
 		c.Assert(r.Found, Equals, true)
 		c.Assert(r.Source["es_title"], Equals, "hello")
 	}
+
+	testBuildTable(c)
```
Comment (siddontang) [inline river/river_test.go]: move these to another independent test, like

```
func TestTableValidation(t *testing.T) {
     tbl := []struct {
         Tables []string,
         Expect bool,
     }

    for _, t := range tbl {
       if !isValidTables(t.Tables), Expect) {
            t.Errrof()
        }
   }
}
```

The same for table build.

**Comment by perfectacle** [comment]
@siddontang hey... I separated the test code... plz check it haha

File: river/river_test.go [hunk 5]
```diff
@@ -25,6 +25,12 @@ type riverTestSuite struct {
 	r *River
 }
 
+type Assertion struct {
```
Comment (siddontang) [inline river/river_test.go]: don't use this, using the anonymous struct is better.

**Comment by perfectacle** [comment]
@siddontang I use anonymous struct, plz chk it~

File: river/river_test.go [hunk 6]
```diff
@@ -351,3 +351,36 @@ func (s *riverTestSuite) TestRiver(c *C) {
...
+		Expect bool
+	} {
+		{[]string{"*"}, true},
+		{[]string{"table", "table2"}, true},
+		{[]string{"*", "table"}, false},
+	}
+
+	for _, table := range tables {
+		if isValidTables(table.Tables) != table.Expect {
+			t.Errorf("\nTables: %s\nExpected: is %t\nbut: was %t\n", table.Tables, table.Expect, isValidTables(table.Tables))
```
Comment (siddontang) [inline river/river_test.go]: please remove all `\n`

File: river/river_test.go [hunk 7]
```diff
@@ -351,3 +351,36 @@ func (s *riverTestSuite) TestRiver(c *C) {
...
+		Table string
+		Expect string
+	} {
+		{"*", ".*"},
+		{"table2", "table2"},
+	}
+
+	for _, table := range tables {
+		if buildTable(table.Table) != table.Expect {
+			t.Errorf("\nTable: %s\nExpected: is \"%s\"\nbut: was \"%s\"\n", table.Table, table.Expect, buildTable(table.Table))
```
Comment (siddontang) [inline river/river_test.go]: ditto

File: river/river_test.go (hunk 6 above)
Comment (perfectacle) [reply river/river_test.go]: all \n...?
you mean write 1 line...??

**Comment by perfectacle** [comment]
I replaced line break to comma, is it right?

**Pull request merged**